src/user/supla_esp_devconn.c

```diff
@@ -1589,3 +1699,136 @@ void DEVCONN_ICACHE_FLASH supla_esp_calcfg_result(TDS_DeviceCalCfgResult *result
+    unsigned char channel_number;
+	char value[SUPLA_CHANNELVALUE_SIZE];
+    TElectricityMeter_ExtendedValue ev;
+    TElectricityMeter_Value v;
+
+    memset(&ev, 0, sizeof(TElectricityMeter_ExtendedValue));
+	memset(&v, 0, sizeof(TElectricityMeter_Value));
+
+	channel_number = 1;
+    memset(value, 0, sizeof(SUPLA_CHANNELVALUE_SIZE));
+	supla_getVoltage(value);
+	memcpy(&napiecie, value, sizeof(uint32_t));
+	last_voltage = (int)(napiecie);
+	if ( relay_laststate == 0) last_voltage = 0;
+	supla_log(LOG_DEBUG, "Voltage: %i", last_voltage);
+    memset(value, 0, sizeof(SUPLA_CHANNELVALUE_SIZE));
+	supla_getCurrent(value);
+	memcpy(&prad, value, sizeof(_supla_int64_t));
+	last_current = (int)(prad);
+	if ( relay_laststate == 0 ) last_current = 0;
+	if ( abs(last_dif_current - last_current) > 0) {
+		if ( last_current >= last_dif_current )  current_difference = (int)(((last_current - last_dif_current)*100)/last_current);
+		else  current_difference = (int)(((last_dif_current - last_current)*100)/last_dif_current);
+		if ( current_difference >= 10 ) {
+			snd_current = 1;
+		}
+	}
+	supla_log(LOG_DEBUG, "Current: %i", last_current);
+	last_dif_current = last_current;
+    memset(value, 0, sizeof(SUPLA_CHANNELVALUE_SIZE));
+	supla_getPower(value);
+	memcpy(&power, value, sizeof(_supla_int64_t));
+	last_power = (int)(power);
+	if ( relay_laststate == 0 ) last_power = 0;
+	supla_log(LOG_DEBUG, "Power: %i", last_power);
+	sekundy = (uint32_t)(sntp_get_current_timestamp());
+	time_difference = sekundy - last_seconds;
+	last_seconds = sekundy;
+	if ( time_difference == 0 ) time_difference = MEASUREMENT_TIME;
+	if ( time_difference > 10*MEASUREMENT_TIME )  time_difference = MEASUREMENT_TIME;
+	if (last_power*time_difference > 0) {
+		supla_esp_state.full_energy = supla_esp_state.full_energy + last_power*time_difference;
+		supla_esp_save_state(0);
+	}
+	if ( abs(last_dif_power - last_power) > 0) {
+		if ( last_power >= last_dif_power )  power_difference = (int)(((last_power - last_dif_power)*100)/last_power);
+		else  power_difference = (int)(((last_dif_power - last_power)*100)/last_dif_power);
+	}
+	last_dif_power = last_power;
+	supla_log(LOG_DEBUG, "power_difference: %i, last_power: %i W, last_dif_power: %i W, interval: %i sec", power_difference, last_power, last_dif_power, time_difference);
+	
+    v.flags = EM_VALUE_FLAG_PHASE1_ON;
+	v.total_forward_active_energy = (unsigned int)supla_esp_state.full_energy/36000;
+    memcpy(value, &v, sizeof(TElectricityMeter_Value));
+    supla_esp_channel_value__changed(channel_number, value);
+	
+    ev.total_forward_active_energy[0]   = supla_esp_state.full_energy/36;
+
+	ev.m->voltage[0] = last_voltage*10;
+	ev.m->current[0] = last_current;
+	ev.m->power_active[0] = last_power*100000;
+	
+	ev.measured_values = EM_VAR_VOLTAGE | EM_VAR_CURRENT | EM_VAR_POWER_ACTIVE | EM_VAR_FORWARD_ACTIVE_ENERGY;
+	ev.m_count = 1;
+	ev.period  = 0;
+
+    supla_esp_channel_em_value_changed(channel_number, &ev);
+}
+
```

Comment:
The function implementation is missing
void ICACHE_FLASH_ATTR supla_esp_em_extendedvalue_to_value (
     TElectricityMeter_ExtendedValue * ev, char * value) {
}